### DISCUSSION TITLE
Turborepo 2.x memory leaks from underlying tools?

### DISCUSSION THREAD

**Discussion by wujekbogdan** [vercel/turborepo#8500] [Help]
### Summary

I updated from `1.13.4` to `2.0.3` lately, without making any other changes, and since then my CI pipelines keep failing with the `ELIFECYCLE` error code and the _"exceeded memory limit"_ error message.

Reverting to `1.13.4` makes the issue go away.

I'm not reporting a bug yet because I'm not really sure what's going on. I wanted to check first if I'm the only one having this issue after the Turbo 2.x upgrade.

### Additional information

- node v18.14.2
- ~25 rather small TS packages
- TypeScript 4.9.5
- No bundler; the code gets compiled with `tsc`
- It's the `turbo test` command that fails. The failure is not related to any unit test in particular; it happens when building packages because `test` depends on `^build`.

**Comment by anthonyshew** [comment]
If you use `turbo <your-tasks> --env-mode=loose` on your command, do you get a different result?

**Reply by wujekbogdan** [reply]
Yes, setting that flag resolves the issue. I toggled it on an off several times and the results are consistent.

**Reply by anthonyshew** [reply]
Which testing tool are you using?

**Reply by wujekbogdan** [reply]
Jest 29.5.0
